plugins/microservices/msi_get_open_data_obj_l1desc_index/libmsi_get_open_data_obj_l1desc_index.cpp

```diff
@@ -0,0 +1,82 @@
+/// \file
+
+#include "irods_ms_plugin.hpp"
+#include "irods_re_structs.hpp"
+#include "msParam.h"
+#include "rodsErrorTable.h"
+#include "irods_error.hpp"
+#include "irods_log.hpp"
+#include "objDesc.hpp"
+#include "fileOpr.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+// persistent L1 object descriptor table
+extern l1desc_t L1desc[NUM_L1_DESC];
+
+namespace
+{
+    int msi_impl(msParam_t* _in, msParam_t* _out, ruleExecInfo_t* _rei)
+    {
+        // check sanity
+        if (!_in || !_out) {
+            return SYS_INTERNAL_ERR;
```

Comment:
Consider using a different error code.

SYS_INVALID_INPUT_PARAM?

Reply:
I was thinking that too, but it seems that the RE should always initialize the pointers for the msParam_t structs, and was thinking that maybe the best way to relay an error w/ memory allocation would be `SYS_INTERNAL_ERR`?

Reply:
Used Alan's suggestion SYS_INTERNAL_NULL_INPUT_ERR, committed.